### DISCUSSION TITLE
Access NER Tags in a converted .spacy file

### DISCUSSION THREAD

**Discussion by fariiaakh** [explosion/spaCy#11496] [Help: Coding & Implementations] [answered]
Hi Everyone,

Raphael and Thomas, who are a few of the SPACY developers helped me to convert my IOB file to a .spacy file.

However, when I asked about how I can view the NER tags in my .spacy converted file, I was told to use the '.label_' property.
https://github.com/explosion/spaCy/discussions/11391#discussioncomment-3606010

The .label_ property works for accessing entities using pretrained spacy models. I want to train a model with my converted data but I want to see if the data converted properly first but I am unable to see that:

![image](https://user-images.githubusercontent.com/55149721/189984724-9643e078-66aa-4796-bf3a-f635b8a3cfe4.png)

I know the '.label_ ' property works for doc objects, based on the spacy course, but what i have is a DocBin object and there is no property for extracting the entity label. 

I converted and IOB file to a .spacy file. 

This is what the iob file looks like:
![image](https://user-images.githubusercontent.com/55149721/189985683-f563dadf-23e6-4545-83f0-0fcb8bb79726.png)

When i converted my IOB file to json, i was able to see what the data looks like: 
![image](https://user-images.githubusercontent.com/55149721/189985537-0ecf9e6f-debf-4fa9-a9f1-03a94aedac1d.png)

But i cannot see what my converted spacy file looks like. I can access the vocab with nlp.vocab but not the ner tags. 

Please help!

**Comment by richardpaulhudson** [comment, marked as answer]
- Please don't paste images of code or terminal output; paste text as text. It's hard for us to read images and we can't modify your code or inspect, and we can't search it on the forums. You could have a look at [this link](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) that offers general guidelines about submitting issues and discussions on Github.

- I think what is perhaps a bit confusing here is that there are two views of the same entity data. You can set and access entity labels either at the `Doc` level e.g. using `set_ents()` or at the level of the individual tokens that make up a `Doc`. For example, `token = doc[2]` returns you the third token in the document. You can then access any entity information relating to that token via `token.ent_type_` and `token.ent_iob_`: this is just a different view of the same information you set using `set_ents()`. See [this link](https://spacy.io/usage/linguistic-features#accessing-ner) for more information.

- When you create a `DocBin` object, you can specify a list of token attributes to include for each document. The attribute labels for the information you are interested in here would be `ENT_TYPE` and `ENT_IOB`, so you could create a `DocBin` with `doc_bin = DocBin(attrs=["ENT_IOB", "ENT_TYPE"])`. However, you can also leave the `attrs` parameter out altogether, in which case a `DocBin` is created that stores a number of standard parameters that includes these two. See [this link](https://spacy.io/api/docbin#init).

- When you retrieve the documents from a `DocBin`, you always specify the `Vocab` object from the pipeline with which the `DocBin` was created (see the example [here](https://spacy.io/api/docbin#get_docs)). Once you've done that and got each document, you can access the entity information within a document using either the document-based or the token-based view.

**Reply by fariiaakh** [reply]
Thank You So Much Richard!! I've been stuck on this for weeks!! Thank You!!
